src/cow-react/modules/wallet/web3-react/connection/ledger.tsx

```diff
@@ -0,0 +1,45 @@
+import { Connector } from '@web3-react/types'
+import { ConnectionType } from '@cow/modules/wallet'
+import { getConnectionName } from '@cow/modules/wallet/api/utils/connection'
+import { useIsActiveWallet } from 'hooks/useIsActiveWallet'
+
+import { ConnectWalletOption } from '@cow/modules/wallet/api/pure/ConnectWalletOption'
+import { default as LedgerImage } from '@cow/modules/wallet/api/assets/ledger.svg'
+import { Ledger } from '../connectors/LedgerConnector'
+import { initializeConnector } from '@web3-react/core'
+import { Web3ReactConnection } from '../types'
+import { RPC_URLS } from 'constants/networks'
+
+const BASE_PROPS = {
+  color: '#4196FC',
+  icon: LedgerImage,
+  id: 'ledger',
+}
+
+const [ledger, ledgerHooks] = initializeConnector<Ledger>(
+  (actions) =>
+    new Ledger({
```

Comment:
Please, make it lazy-loading

Reply:
It doesn't work properly with AsyncConnector https://github.com/cowprotocol/cowswap/pull/2227
The balances are not loaded on refresh and if you disconnect it will constantly open ledger-live modal on every refresh. Its probably because Async connector is only using activate instead of `connectEagerly` but I think this is something we can fix in the future.

Reply:
But why it works with Trezor?
Anyway, you can add lazy loading inside of Ledger connector class

Reply:
We don't use `connectEagerly` on Trezor